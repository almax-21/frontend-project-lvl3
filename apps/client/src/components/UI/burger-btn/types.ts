export interface BurgerBtnProps {
	isActive: boolean;
	onClick: () => void;
}
