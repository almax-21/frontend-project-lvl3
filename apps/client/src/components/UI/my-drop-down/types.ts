import type { Variant } from 'react-bootstrap/esm/types';

export interface MyDropDownProps<T> {
	variant: Variant;
	title: string;
	values: Array<T>;
	activeValue: T;
	handleSetActiveValue: (value: T) => () => void;
}
