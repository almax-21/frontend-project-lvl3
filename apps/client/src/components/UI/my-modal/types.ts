export enum MODAL_TYPES {
	PREVIEW = 'PREVIEW',
	MARK = 'MARK',
	DELETE = 'DELETE',
	RELOAD = 'RELOAD',
}

export type MODAL_TYPE = typeof MODAL_TYPES[keyof typeof MODAL_TYPES];
