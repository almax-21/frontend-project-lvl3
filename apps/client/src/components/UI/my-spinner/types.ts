export interface MySpinnerProps {
	small?: boolean;
	isDark?: boolean;
}
