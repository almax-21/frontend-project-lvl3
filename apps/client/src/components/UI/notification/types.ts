import type { Variant } from 'react-bootstrap/esm/types';

export enum NOTIFICATION_VARIANT {
	SUCCESS = 'success',
	ERROR = 'danger',
}

export interface NotificationData {
	variant: Variant;
	message: string;
}
