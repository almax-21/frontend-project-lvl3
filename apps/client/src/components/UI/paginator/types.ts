export enum PAGINATOR_CAPACITY {
	MIN = 2,
	EXTRA = 8,
	MIDPOINT = 5,
	RANGE = 3,
}

export interface PaginationProps {
	pages: number[];
	activePage: number;
	onClick: (page: number) => () => void;
}
