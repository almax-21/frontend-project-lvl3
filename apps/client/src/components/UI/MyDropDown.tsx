import React, { FC } from 'react';
import { Dropdown, DropdownButton } from 'react-bootstrap';
import { Variant } from 'react-bootstrap/esm/types';
import { FormattedMessage } from 'react-intl';

import useTypedSelector from '../../hooks/redux/useTypedSelector';
import { selectSettings } from '../../store/selectors/settingsSelectors';
import { POST_STATE_TYPE } from '../../store/types';

interface MyDropDownProps<T> {
	variant: Variant;
	title: string;
	values: Array<T>;
	activeValue: T;
	handleSetActiveValue: (value: T) => () => void;
}

const MyDropDown: FC<MyDropDownProps<POST_STATE_TYPE>> = ({
	variant,
	title,
	values,
	activeValue,
	handleSetActiveValue,
}) => {
	const { isDarkTheme } = useTypedSelector(selectSettings);

	return (
		<DropdownButton
			menuVariant={isDarkTheme ? 'dark' : undefined}
			title={title}
			variant={variant}
		>
			{values.map((value) => (
				<Dropdown.Item
					key={value}
					active={value === activeValue}
					onClick={handleSetActiveValue(value)}
				>
					<FormattedMessage id={value} />
				</Dropdown.Item>
			))}
		</DropdownButton>
	);
};

export default MyDropDown;
