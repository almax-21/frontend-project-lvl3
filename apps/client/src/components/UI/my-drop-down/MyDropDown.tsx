import type { FC } from 'react';
import React from 'react';
import { Dropdown, DropdownButton } from 'react-bootstrap';
import { FormattedMessage } from 'react-intl';

import { useTypedSelector } from '@/hooks';
import { selectSettings } from '@/store/selectors/settingsSelectors';
import type { POST_STATE_TYPE } from '@/store/types';

import type { MyDropDownProps } from './types';

export const MyDropDown: FC<MyDropDownProps<POST_STATE_TYPE>> = ({
	variant,
	title,
	values,
	activeValue,
	handleSetActiveValue,
}) => {
	const { isDarkTheme } = useTypedSelector(selectSettings);

	return (
		<DropdownButton
			menuVariant={isDarkTheme ? 'dark' : undefined}
			title={title}
			variant={variant}
		>
			{values.map((value) => (
				<Dropdown.Item
					key={value}
					active={value === activeValue}
					onClick={handleSetActiveValue(value)}
				>
					<FormattedMessage id={value} />
				</Dropdown.Item>
			))}
		</DropdownButton>
	);
};
