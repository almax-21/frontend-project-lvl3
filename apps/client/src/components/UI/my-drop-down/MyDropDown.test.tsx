import type { ComponentType } from 'react';
import React from 'react';
import { compose } from '@reduxjs/toolkit';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import withIntl from '@/hocs/withIntl';
import withStateProvider from '@/hocs/withStateProvider';
import settingsReducer from '@/store/slices/settingsSlice';
import type { POST_STATE_TYPE } from '@/store/types';
import { POST_STATES } from '@/store/types';

import type { MyDropDownProps } from './MyDropDown';
import { MyDropDown } from './MyDropDown';

it('should toggle drop-down on click', async () => {
	const MyDropDownSmart = compose<
		ComponentType<MyDropDownProps<POST_STATE_TYPE>>
	>(
		withStateProvider({ settings: settingsReducer }),
		withIntl,
	)(MyDropDown);

	const values = Object.values(POST_STATES);

	const { container } = render(
		<MyDropDownSmart
			activeValue={POST_STATES.ALL}
			title={'LOL'}
			values={values}
			variant="outline-secondary"
			onItemClick={jest.fn()}
		/>,
	);

	expect(container).toMatchSnapshot();

	const toggleEl = screen.getByRole('button', { name: 'LOL', expanded: false });
	await userEvent.click(toggleEl);

	expect(container).toMatchSnapshot();
});
