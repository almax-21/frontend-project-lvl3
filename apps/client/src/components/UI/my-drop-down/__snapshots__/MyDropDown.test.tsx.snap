// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should toggle drop-down on click 1`] = `
<div>
  <div
    class="dropdown"
  >
    <button
      aria-expanded="false"
      class="dropdown-toggle btn btn-outline-secondary"
      type="button"
    >
      LOL
    </button>
  </div>
</div>
`;

exports[`should toggle drop-down on click 2`] = `
<div>
  <div
    class="show dropdown"
  >
    <button
      aria-expanded="true"
      class="dropdown-toggle btn btn-outline-secondary"
      type="button"
    >
      LOL
    </button>
    <div
      aria-labelledby=""
      class="dropdown-menu show"
      data-popper-escaped="true"
      data-popper-placement="bottom-start"
      data-popper-reference-hidden="true"
      style="position: absolute; top: 0px; left: 0px; transform: translate(0px, 2px);"
      x-placement="bottom-start"
    >
      <a
        aria-selected="false"
        class="dropdown-item"
        data-rr-ui-dropdown-item=""
        href="#"
        role="button"
        tabindex="0"
      >
        Read
      </a>
      <a
        aria-selected="false"
        class="dropdown-item"
        data-rr-ui-dropdown-item=""
        href="#"
        role="button"
        tabindex="0"
      >
        Unread
      </a>
      <a
        aria-selected="true"
        class="dropdown-item active"
        data-rr-ui-dropdown-item=""
        href="#"
        role="button"
        tabindex="0"
      >
        All
      </a>
    </div>
  </div>
</div>
`;
